core/templates/dev/head/domain/exploration/StateTopAnswerStatisticsFactory.js

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for domain object which holds the list of top answer
+ * statistics for a particular state.
+ */
+
+oppia.factory('StateTopAnswerStatisticsFactory', [
+  '$http', '$injector', 'AngularNameService', 'AnswerClassificationService',
+  'ExplorationContextService', 'ExplorationStatesService',
+  function(
+      $http, $injector, AngularNameService, AnswerClassificationService,
+      ExplorationContextService, ExplorationStatesService) {
+    /**
+     * @typedef {Object} AnswerStatistics - A record for the statistics of a
+     * single top answer.
+     *
+     * @property {*} answer - Contains the answer in its raw form, directly from
+     *    the backend.
+     * @property {string} answerAsHtml - Contains the answer as a string which
+     *    can be rendered directly as HTML.
+     * @property {number} frequency
+     * @property {boolean} isAddressed
+     */
+
+    /**
+     * TODO(brianrodri): Move this helper function into a proper service which
+     * takes the state's interaction type into account when formatting the HTML.
+     * @param {*} answer
+     * @returns {string}
+     */
+    var convertAnswerToHtml = function(answer) {
+      // Don't render quotes when the answer is a string.
+      return (typeof answer === 'string') ? answer : JSON.stringify(answer);
+    };
+
+    /**
+     * @private @constructor
+     * Returns a state's top answers with stale data. For a fresh instance, use
+     * {@link StateTopAnswerStatistics#createFromBackendDict} instead, or call
```

Comment:
Not sure about the "instead" here since that method actually calls this one. Maybe just:

Returns a state's top answers with stale data. This needs to be refreshed by calling ... after construction.